runtime/onert/core/src/util/EventWriter.h

```diff
@@ -128,7 +128,7 @@ class EventWriter
   void flush(WriteFormat write_format);
 
 private:
-  static std::mutex _mutex;
+  static inline std::mutex _mutex;
```

Comment:
@ragmani Is it OK to use static inline for mutex class variable? Doesn't it have possibility to inlining mutex object by compiler?

Reply:
Yes, it's perfectly fine to use static inline for a mutex class variable. The inline keyword in this context is used to ensure that the variable has a single definition across all translation units, which allows you to define it in a header file without causing multiple definition errors. It does not cause the mutex object itself to be "inlined" or optimized away in the way that function inlining might; it merely affects linkage. The mutex will still behave as expected for synchronization.

Reply:
I found answer at https://en.cppreference.com/w/cpp/language/storage_duration#external_linkage

>- Any of the following names declared at namespace scope have external linkage, unless they are declared in an unnamed namespace
>   - names of classes, their member functions, **static data members (const or not)**, nested classes and enumerations, and functions first introduced with [friend](https://en.cppreference.com/w/cpp/language/friend) declarations inside class bodies;

Reply:
This feature is hard to understand based on compiler world...

Reply:
> This feature is hard to understand based on compiler world...

The two types of variables(static class member or namespace-scope variable) can be declared as inline variables. If both variables associated with inline keyword are problematic, it can be expected that the compiler will generate errors about that.
I think there is no need to overthink it and we just follows the rule at https://en.cppreference.com/w/cpp/language/inline regardless of compiler world.
> Because the meaning of the keyword inline for functions came to mean "multiple definitions are permitted" rather than "inlining is preferred" since C++98, that meaning was extended to variables.

Reply:
Inline variables (introduced in C++17) help avoid multiple definitions across translation units by ensuring a single instance is shared. Inline variables don't change the rules of static initialization. They only help ensure that there is a single definition across translation units. So, if some static variables are already safe from the initialization order fiasco, switching them to inline variables won't affect that safety.